### ISSUE TITLE
clang-cl builds fail generating AST files

### ISSUE THREAD

**Issue by bodand** [bodand/cg3#23]
Check tests rely on generating AST files using the build clang, however clang-cl builds fail while generating these files.
Clang-cl reports undefined standard types as seen below.
Without looking into it further now, when building with `clang` in the custom targets, the main compiler also being something-like clang causes a disturbance, but this could be way off.

```
  Generating data/bad.cxx.ast
  In file included from D:/.projects/cg3-clone/checks/arityck/test/data/bad.cxx:12:
  In file included from C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\string:9:
  In file included from C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\yvals_core.h:436:
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h(386,51): error GC871EEFB: unknown type name 'uintptr_t'; did you mean 'intptr_t'? [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
          void __cdecl __security_check_cookie(_In_ uintptr_t _StackCookie);
                                                    ^
  C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h:195:30: note: 'intptr_t' declared here
      typedef __int64          intptr_t;
                               ^
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h(387,67): error GC871EEFB: unknown type name 'uintptr_t'; did you mean 'intptr_t'? [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
          __declspec(noreturn) void __cdecl __report_gsfailure(_In_ uintptr_t _StackCookie);
                                                                    ^
  C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h:195:30: note: 'intptr_t' declared here
      typedef __int64          intptr_t;
                               ^
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h(391,8): error GC871EEFB: unknown type name 'uintptr_t'; did you mean 'intptr_t'? [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
  extern uintptr_t __security_cookie;
         ^
  C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h:195:30: note: 'intptr_t' declared here
      typedef __int64          intptr_t;
                               ^
  In file included from D:/.projects/cg3-clone/checks/arityck/test/data/bad.cxx:12:
  In file included from C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\string:11:
  In file included from C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring:13:
  In file included from C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\iosfwd:9:
  In file included from C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\yvals.h:17:
  In file included from C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\crtdbg.h:12:
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt.h(380,16): error GC871EEFB: unknown type name 'uintptr_t'; did you mean 'intptr_t'? [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
      _In_       uintptr_t _Reserved);
                 ^
  C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\vcruntime.h:195:30: note: 'intptr_t' declared here
      typedef __int64          intptr_t;
                               ^
  In file included from D:/.projects/cg3-clone/checks/arityck/test/data/bad.cxx:12:
  In file included from C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\string:11:
  In file included from C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring:13:
  In file included from C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\iosfwd:12:
  In file included from C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\cstdio:12:
  In file included from C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\stdio.h:13:
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(277,49): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                                  va_list          _ArgList
                                                  ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(286,49): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                                  va_list          _ArgList
                                                  ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(295,49): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                                  va_list          _ArgList
                                                  ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(303,49): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                                  va_list              _ArgList
                                                  ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(317,39): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                        va_list              _ArgList
                                        ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(332,49): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                                  va_list              _ArgList
                                                  ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(348,43): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                            va_list              _ArgList
                                            ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(365,49): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                                  va_list              _ArgList
                                                  ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(379,39): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                        va_list              _ArgList
                                        ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(393,49): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                                  va_list              _ArgList
                                                  ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(406,39): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                        va_list              _ArgList
                                        ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(420,49): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                                  va_list              _ArgList
                                                  ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(435,43): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                            va_list              _ArgList
                                            ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(451,49): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                                  va_list              _ArgList
                                                  ^
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\corecrt_wstdio.h(464,39): error GC871EEFB: unknown type name 'va_list' [D:\.projects\cg3-clone\_build-release-x64-clang\checks\arityck\test\cg3-check-arityck-test-data.vcxproj]
                                        va_list              _ArgList
                                        ^
```

**Comment by bodand** [comment]
Include path problems, trivial fix is to just don't include things, and for the C++ files, use a `bool` variable. C23 got it afaik as well, but we all know C people are not going to use it for at least half a century.

**Comment by bodand** [comment]
fix in 1c72bdf3
